[1.2.0] panic: runtime error: slice bounds out of range [13:12]

```
panic: runtime error: slice bounds out of range [13:12]

goroutine 7 [running]:
github.com/buger/goreplay/proto.HasResponseTitle(0xc00027ec66, 0xb7, 0xb7, 0xc002610200)
	/go/src/github.com/buger/goreplay/proto/proto.go:357 +0x242
main.startHint(0xc002c12370, 0xc0003a0630)
	/go/src/github.com/buger/goreplay/input_raw.go:203 +0x7c
github.com/buger/goreplay/tcp.(*MessagePool).Handler(0xc0003a2a40, 0x1124fc0, 0xc003d5da20)
	/go/src/github.com/buger/goreplay/tcp/tcp_message.go:176 +0x9fa
github.com/buger/goreplay/capture.(*Listener).Listen(0xc00034f4a0, 0x1118860, 0xc0003a2a80, 0xc000380400, 0x0, 0x0)
	/go/src/github.com/buger/goreplay/capture/capture.go:163 +0x163
github.com/buger/goreplay/capture.(*Listener).ListenBackground.func1(0xc00011efc0, 0xc00034f4a0, 0x1118860, 0xc0003a2a80, 0xc000380400)
	/go/src/github.com/buger/goreplay/capture/capture.go:173 +0x7b
created by github.com/buger/goreplay/capture.(*Listener).ListenBackground
	/go/src/github.com/buger/goreplay/capture/capture.go:171 +0x84
```

--- a/proto/proto.go
+++ b/proto/proto.go
@@ -21,7 +21,6 @@ import (
 	"bytes"
 	"net/http"
 	"net/textproto"
-	"strconv"
 	"strings"
 
 	"github.com/buger/goreplay/byteutils"
@@ -344,8 +343,11 @@ func HasResponseTitle(payload []byte) bool {
 	if !(ok && major == 1 && (minor == 0 || minor == 1)) {
 		return false
 	}
-	status, err := strconv.Atoi(s[VersionLen+1 : VersionLen+4])
-	if err != nil {
+	if s[VersionLen] != ' ' {
+		return false
+	}
+	status, ok := atoI(payload[VersionLen+1:VersionLen+4], 10)
+	if !ok || s[VersionLen+4] != ' ' {
 		return false
 	}
 	statusText := http.StatusText(status)
@@ -484,7 +486,7 @@ func atoI(s []byte, base int) (num int, ok bool) {
 			return 0, false
 		}
 		v = int(hexTable[s[i]])
-		if v >= base {
+		if v >= base || (v == 0 && s[i] != '0') {
 			return 0, false
 		}
 		num = (num * base) + v

--- a/proto/proto_test.go
+++ b/proto/proto_test.go
@@ -330,7 +330,9 @@ func TestHasResponseTitle(t *testing.T) {
 		"HTTP/1.1 100 Continue\r\n": true,
 		"HTTP/1.1  \r\n":            false,
 		"HTTP/4.0 100Continue\r\n":  false,
-		"HTTP/4.0 100 Continue\r\n": false,
+		"HTTP/1.0 10r Continue\r\n": false,
+		"HTTP/1.1 200\r\n":          false,
+		"HTTP/1.1 200\r\nServer: Tengine\r\nContent-Length: 0\r\nConnection: close\r\n\r\n": false,
 	}
 	for k, v := range m {
 		if HasResponseTitle([]byte(k)) != v {

